katdal/lazy_indexer.py

```diff
@@ -20,6 +20,52 @@
 
 # TODO support advanced integer indexing with non-strictly increasing indices (i.e. out-of-order and duplicates)
 
+
+def _simplify_index(shape, indices):
+    """Generate an equivalent index expression that is cheaper to evaluate.
+
+    In the current implementation, boolean numpy arrays which select contiguous
+    ranges are converted to slices. Note that when indexing along multiple axes
+    with arrays, this may change the semantics of the indexing (see this `NEP`_
+    for details).
+
+    .. _NEP: https://gist.github.com/seberg/976373b6a2b7c4188591
+
+    If the expression is invalid in some way (e.g. wrong length array index)
+    the original index is returned so that that an exception will be raised
+    when the index is used.
+
+    Ellipses are currently not supported, and will result in the original
+    index being returned.
+    """
+    if not isinstance(indices, tuple):
+        indices = (indices,)
+    out = []
+    axis = 0
+    for index in indices:
+        if index is Ellipsis:
+            return indices     # Not supported
+        elif index is np.newaxis:
+            out.append(index)
```

Comment:
Do you also need to increment `axis` here? If so, I would redesign the loop to have a single `out.append(index)` at the end.